util/recorder.js

```diff
@@ -0,0 +1,952 @@
+import fs from "fs";
+import path from "path";
+import os from "os";
+
+import { v4 as uuidv4 } from "uuid";
+
+import PQueue from "p-queue";
+
+import { logger, errJSON } from "./logger.js";
+import { sleep, timestampNow } from "./timing.js";
+import { RequestResponseInfo } from "./reqresp.js";
+
+import { baseRules as baseDSRules } from "@webrecorder/wabac/src/rewrite/index.js";
+import { rewriteDASH, rewriteHLS } from "@webrecorder/wabac/src/rewrite/rewriteVideo.js";
+
+import { WARCRecord } from "warcio";
+import { WARCSerializer } from "warcio/node";
+import { WARCWriter } from "./warcwriter.js";
+
+const MAX_BROWSER_FETCH_SIZE = 2_000_000;
+const MAX_NETWORK_LOAD_SIZE = 200_000_000;
+
+const ASYNC_FETCH_DUPE_KEY = "s:fetchdupe";
+
+const WRITE_DUPE_KEY = "s:writedupe";
+
+const encoder = new TextEncoder();
+
+// =================================================================
+function logNetwork(/*msg, data*/) {
+  // logger.debug(msg, data, "recorderNetwork");
```

Comment:
For now I'd suggest leaving it in but functional

Reply:
yes, need to make sure that this context is not enabled by default, though. We should probably set a default debugging context so that its not everything..